### DISCUSSION TITLE
Finding which of two nodes is closer to a focal node in a tree

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/tskit#2654] [Q&A] [answered]
If I have a focal node and two target nodes in a tree, what's the neatest way to find out which of the two target nodes is the closest relative? Perhaps look at the tMRCAs against both?

**Comment by jeromekelleher** [comment, marked as answer]
Yes, using the MRCA is the simplest approach I think. You would need to decide whether overall branch length or path length (number of hops) is your metric.

**Reply by hyanwong** [reply]
Assuming I mean closest topologically, I think it's actually the raw tMRCA that I need:

```
def topologically_closest(tree, focal, targets):
    """
    Return the indices into targets indicating which have the smallest tMRCA
    against focal in the given tree. Because tree nodes are strictly time ordered,
    this is also guaranteed to be the list of targets which are topologically closest.
    """
    tmrcas = np.array([tree.tmrca(focal, t) for t in targets])
    min_tmrca = tmrcas.min()
    return np.where(min_tmrca == trmcas)[0]
```
